src/task.c

```diff
@@ -438,7 +438,7 @@ void JL_NORETURN throw_internal(jl_value_t *exception JL_MAYBE_UNROOTED)
         jl_longjmp(*ptls->safe_restore, 1);
     JL_GC_PUSH1(&exception);
     jl_gc_unsafe_enter(ptls);
-    if (exception) {
+    if (exception && (ptls->bt_size != 0)) {
```

Comment:
I guess the exception stack should be able to handle the case of `ptls->bt_size == 0` and still preserve a non-null exception for later analysis.

Reply:
(If it can't handle `bt_size == 0` that's a bug and should be fixed.)